### DISCUSSION TITLE
Audio Download Quality

### DISCUSSION THREAD

**Discussion by Sailfish** [audiamus/BookLibConnect#27] [Ideas]
The Audio Download Quality selections "Low" and "Normal" both appear to be monaural and on closer inspection appear to be the same waveforms (see BookLibConnectQuality(1_LOW).jpg and BookLibConnectQuality(1_Normal).jpg); whereas, BookLibConnectQuality(1_HIGH).jpg is clearly stereo. Would there be a room to add the monaural(mono) and stereo qualifier in the button dropdown? See attached screenshots.

Also, for the monaural .aax file, it only plays in one headphone speaker; although, the *.m4b file plays in both.

![BookLibConnectQuality(1_LOW)](https://user-images.githubusercontent.com/110050/155256802-7103f1fb-c45a-4fc1-a0df-16ce562c77f3.jpg)

![BookLibConnectQuality(2_NORMAL)](https://user-images.githubusercontent.com/110050/155256806-69095cea-09f3-464d-8110-436a47b525db.jpg)
![BookLibConnectQuality(3_HIGH)](https://user-images.githubusercontent.com/110050/155256810-57796f2c-e74c-4393-a596-d215b1a1f09b.jpg)

**Comment by audiamus** [comment]
The audio quality you request from Audible is given as an abstract value. The concrete response is up to Audible and appears to be very much dependent on the individual book. Audible can play with the number of channels, the sampling rate and the bit rate:

- Number of channels: mono/stereo.
- Sampling rate: the conversion between analogue and digital audio signals. The highest sampling rate observed is 44.1 kHz, which is CD quality. The highest reproducible analogue frequency would be 22.05 kHz, see [Nyquist–Shannon](https://en.wikipedia.org/wiki/Nyquist%E2%80%93Shannon_sampling_theorem).
- Bit rate: Audible AAX and AAXC files are MP4 containers with a compressed audio stream. The audio codec is AAC, the standard for MP4 audio. The bit rate affects the compression, and, as AAC is a lossy compression, the audio quality. The highest observed bit rate is 128 kBit/s. Another variation for the bit rate is constant and variable. Audible uses constant bit rate, but does not always indicate it as such.     

As you know, I added functions to lower the audio quality in AAX Audio Audio Converter, to make files shorter during conversion.  With the latest Book Lib Connect version, you can now opt for low audio quality and short files right from the start.

**Reply by Sailfish** [reply]
Thanks for the detailed explanation, it's always very much appreciated.

As I understand your explanation, when selecting Low or Normal quality the decision is left to how Audible formatted the file, i.e., Book Lib Connect has no control over whether a mono or stereo version is produced. For my title archive storage purposes I would prefer to use a lowest sampling rate without appreciable sound quality loss since most titles don't need the High quality for listening to the narration. I know some titles are more elaborate with sound effects an multiple narrators which may dictate using the High quality setting but since I shy away from those titles, it's not an issue for me. So, my concern is when using Book Lib Connect and selecting either Low or Normal, there's a chance that the rendered version would end up being mono (only one headphone speaker active) which is not what I would prefer. Is there a way that Book Lib Connect could provide a setting similar to AAX Audio Audio Converter where one could have control over this? Otherwise, it seems that I would need to go through a double conversion process, i.e., select High using Book Lib Connect and then using  AAX Audio Audio Converter on the resultion .aax file and selecting a lower bit rate there so as to both reduce the titles size and not having it in mono format.

Sorry for the rambling above but I hope I've adequately conveyed my understanding of the concern.

**Reply by audiamus** [reply]
>  Is there a way that Book Lib Connect could provide a setting similar to AAX Audio Audio Converter where one could have control over this?

I'm afraid not. The "Low" to "Extreme" settings are predefined by Audible and the only way to convey the desired audio quality to them. To make sure to get a stereo file I guess you have to select one of the higher settings. You can always scale down with AAX Audio Converter, as you say. And AAX Audio Converter never reduces stereo to mono.

**Comment by Sailfish** [comment]
I understand. Thanks again.
